packages/cli/src/util/dev/server.ts

```diff
@@ -1642,12 +1644,16 @@ export default class DevServer {
         missRoutes,
         phase
       );
-      prevUrl =
```

Comment:
`prevUrl` is no longer assigned `req.url` in each phase. Is that expected?

Reply:
Yes, I believe the bug was caused by the fact that it was being re-set to the original value, before the rewrite was applied.

Here's an example of what would happen:
- given app with
	- "about.html" file 
	- "other.html" file
	- "middleware.js" that will rewrite to "/other.html"
- request to "/about.html", sets `req.url` to "/about.html"
- `prevUrl` is set to "/about.html"
- middleware executed, sets `prevUrl` to "/other.html"
- phase 1 (`null`) runs, doesn't find a route at "/other.html", sets `prevUrl` to `req.url` (THE BUG), which is "/about.html" (NOT the rewrite value of "other.html")
- phase 2 (`filesystem`) runs, sees "about.html", returns it
- note that these are the only 2 `phases` for this logic to loop over

So, we made a request to "/about.html" which the middleware rewrote to "other.html", but we were still served "about.html".

---

The intent of this logic is not super clear, though. I may not have the right of it.